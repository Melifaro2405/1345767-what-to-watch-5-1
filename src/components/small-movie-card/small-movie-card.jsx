import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import MoviePreview from "../movie-preview/movie-preview";
import {filmProptypes} from "../../props-validation";
import {AppRoute} from "../../consts";

const SmallMovieCard = ({film, onMouseOver, onMouseOut, isPlayingVideo}) => {
  const {id, moreInfo, preview} = film;
  const {playPreviewSrc} = moreInfo;
  const {src, title} = preview;

  return (
    <article className="small-movie-card catalog__movies-card">
      <Link className="small-movie-card__link" to={`${AppRoute.FILMS}/${id}`}>
        <div className="small-movie-card__image"
          onMouseOver={onMouseOver}
          onMouseOut={onMouseOut}>
          {isPlayingVideo
            ? <MoviePreview playVideoSrc={playPreviewSrc} src={src} />
            : <img src={src} alt={title} width="280" height="175" />
          }
        </div>
        <h3 className="small-movie-card__title">
          {title}
        </h3>
      </Link>
    </article>
  );
};

SmallMovieCard.propTypes = {
  film: PropTypes.shape(filmProptypes).isRequired,
  onMouseOver: PropTypes.func.isRequired,
  onMouseOut: PropTypes.func.isRequired,
  isPlayingVideo: PropTypes.bool.isRequired
};

export default SmallMovieCard;
