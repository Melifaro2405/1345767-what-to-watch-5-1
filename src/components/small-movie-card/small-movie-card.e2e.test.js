import React from "react";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SmallMovieCard from "./small-movie-card";
import {film} from "./small-movie-card.test";

configure({adapter: new Adapter()});

const noop = () => {};

it(`Should mouse over on small film card`, () => {
  const handleMouseOver = jest.fn();

  const wrapper = shallow(
      <SmallMovieCard
        film={film}
        isPlayingVideo={false}
        onMouseOver={handleMouseOver}
        onMouseOut={noop}
      />
  );

  const smallMovieCard = wrapper.find(`.small-movie-card__image`);
  smallMovieCard.simulate(`mouseover`);
  expect(handleMouseOver).toHaveBeenCalledTimes(1);
});

it(`Should mouse out with small film card`, () => {
  const handleMouseOut = jest.fn();

  const wrapper = shallow(
      <SmallMovieCard
        film={film}
        isPlayingVideo={false}
        onMouseOver={noop}
        onMouseOut={handleMouseOut}
      />
  );

  const smallMovieCard = wrapper.find(`.small-movie-card__image`);
  smallMovieCard.simulate(`mouseout`);
  expect(handleMouseOut).toHaveBeenCalledTimes(1);
});
