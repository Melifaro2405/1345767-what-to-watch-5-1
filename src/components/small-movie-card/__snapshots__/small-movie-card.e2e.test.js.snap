// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should SmallMovieCard render correctly with playing video 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <a
    className="small-movie-card__link"
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <video
        autoPlay={true}
        className="player__video"
        muted={true}
        poster="img/the-grand-budapest-hotel-poster.jpg"
        src="https://some-link"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      The Grand Budapest Hotel
    </h3>
  </a>
</article>
`;

exports[`Should SmallMovieCard render correctly without playing video 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <a
    className="small-movie-card__link"
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="The Grand Budapest Hotel"
        height="175"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      The Grand Budapest Hotel
    </h3>
  </a>
</article>
`;
